src/lib/__tests__/InstantSearch-test.js

```diff
@@ -1007,56 +1024,97 @@ describe('createURL', () => {
       },
     });
 
-    search.addWidget(
-      createWidget({
-        getWidgetState() {
-          return {
-            query: 'Apple',
-          };
+    search.addWidget(connectSearchBox(noop)({}));
+    search.start();
+
+    expect(search.createURL({ indexName: { query: 'Apple' } })).toBe('#');
+  });
+
+  it('returns the default URL for the main index state', () => {
+    const router = createRouter();
+    const search = new InstantSearch({
+      indexName: 'indexName',
+      searchClient: createSearchClient(),
+      initialUiState: {
+        indexName: {
+          query: 'Apple',
         },
-      })
-    );
+      },
+      routing: {
+        router,
+      },
+    });
 
+    search.addWidget(connectSearchBox(noop)({}));
     search.start();
+    search.createURL();
 
-    expect(search.createURL()).toBe('http://algolia.com');
     expect(router.createURL).toHaveBeenCalledWith({
       indexName: {
         query: 'Apple',
       },
     });
   });
 
-  it('returns the URL with the given `SearchParameters` applied', () => {
+  it('returns the URL for nested index states', async () => {
     const router = createRouter();
-    router.createURL.mockImplementation(() => 'http://algolia.com');
-
     const search = new InstantSearch({
       indexName: 'indexName',
       searchClient: createSearchClient(),
+      initialUiState: {
+        indexName: {
+          query: 'Google',
+        },
+        indexNameLvl1: {
+          query: 'Samsung',
+        },
+        indexNameLvl2: {
+          query: 'Google',
+        },
+      },
       routing: {
         router,
       },
     });
 
-    search.addWidget(
+    search.addWidgets([
       createWidget({
```

Comment:
You can use a `connectSearchBox` rather than the mocked version.

Reply:
26cd7a040e01a8602714995cf652927573279834